runtime-tests/index.ts

```diff
@@ -0,0 +1,39 @@
+const config = {
+	chain: 'polkadot',
+};
+const argv = process.argv.slice(0, 2);
+
+/**
+ * We return empty string's as an indication of a false value. JS recognizes "" as falsey.
+ * NodeJS.Process.argv.reduce expects the return value of the accumulator to
+ * be of type string, this is a way of satisfying the compiler, but also
+ * returning the correct behavior.
+ */
+process.argv.reduce((cmd, arg) => {
+	if (cmd) {
+		if (cmd.startsWith('/')) return '';
+		config[cmd] = arg;
+		return '';
+	}
+
+	if (arg.startsWith('--')) {
+		const sub = arg.substring('--'.length);
+		// Will only accept the flag if it's within the config
+		if (Object.keys(config).includes(sub)) {
+			// Set the sub to the next cmd
+			return sub;
+		}
+	}
+
+	argv.push(arg);
```

Comment:
I'm used to seeing `reduce()` return the accumulator at every iteration (including the last) and assign the return value to a variable, e.g. `let sum = [1,2,3].reduce((acc, x) => x*x)`. Instead here it seems like you're not using the return value of the whole `reduce` and relying instead on side-effects like this one. Should it be a for-loop then?

Reply:
So in this case I am using the reducer a little differently than normal, the accumulator in this case is being used to identify falsey values, vs truthy commands. I stayed away from the `for loop` because I didnt want to introduce messy iterator manipulation in order to parse the args, and I think the reducer itself allows for more customization for the future in case we wanted to add arguments to the script.